Basic Search: An unknown error has occurred while performing the search. Contact an administrator if the problem persists. More information available in the logs.

### Issue

I've had some issues with the search, I was able to fix something however, I wasn't able to reproduce it.

First the setup:
I've got a dev and a prod CRM on two different servers. On dev, I've created two modules via the module builder and deployed / module loaded those to prod.
Additionally, I've created studio fields in dev and prod and exported studio customizations over to prod.
And I've edited the `config.php` on prod:
`  'search_wildcard_infront' => true,`

When searching for a complete first name or similar, everything works as expected.
When searching for a part of a word, I see an error in the search.

The logs show:

```
Di. Nov. 25 2025 at 04:03:10 [22610][1][FATAL] [ERROR] [SearchThrowableHandler] An unknown error has occurred while performing the search.
Di. Nov. 25 2025 at 04:03:10 [22610][1][FATAL] [ERROR] [SearchThrowableHandler] TypeError: Unsupported operand types: int + string in /var/www/html/domain/crm.domain/public/legacy/include/database/DBManager.php:1596
Stack trace:
#0 /var/www/html/domain/crm.domain/public/legacy/include/SearchForm/SearchForm2.php(1378): DBManager->quoteType()
#1 /var/www/html/domain/crm.domain/public/legacy/lib/Search/BasicSearch/BasicSearchEngine.php(172): SearchForm->generateSearchWhere()
#2 /var/www/html/domain/crm.domain/public/legacy/lib/Search/BasicSearch/BasicSearchEngine.php(85): SuiteCRM\Search\BasicSearch\BasicSearchEngine->searchModules()
#3 /var/www/html/domain/crm.domain/public/legacy/lib/Search/SearchEngine.php(78): SuiteCRM\Search\BasicSearch\BasicSearchEngine->search()
#4 /var/www/html/domain/crm.domain/public/legacy/lib/Search/SearchWrapper.php(89): SuiteCRM\Search\SearchEngine->searchAndDisplay()
#5 /var/www/html/domain/crm.domain/public/legacy/modules/Home/Search.php(50): SuiteCRM\Search\SearchWrapper::searchAndDisplay()
#6 /var/www/html/domain/crm.domain/public/legacy/include/MVC/View/SugarView.php(800): include_once('...')
#7 /var/www/html/domain/crm.domain/public/legacy/include/MVC/View/views/view.classic.php(72): SugarView->includeClassicFile()
#8 /var/www/html/domain/crm.domain/public/legacy/include/MVC/View/SugarView.php(210): ViewClassic->display()
#9 /var/www/html/domain/crm.domain/public/legacy/include/MVC/Controller/SugarController.php(443): SugarView->process()
#10 /var/www/html/domain/crm.domain/public/legacy/include/MVC/Controller/SugarController.php(374): SugarController->processView()
#11 /var/www/html/domain/crm.domain/public/legacy/include/MVC/SugarApplication.php(101): SugarController->execute()
#12 /var/www/html/domain/crm.domain/public/legacy/index.php(52): SugarApplication->execute()
#13 {main}

```



### Possible Fix

It helped, maybe AI got it right?
I've changed (non upgrade safe):
https://github.com/SuiteCRM/SuiteCRM/blob/40da2845a170832a4e9e9fa0ebe731f8c34de42d/include/database/DBManager.php#L1596


You're getting this error because in PHP 8+, adding an int to a non-numeric string (e.g. "abc") throws a TypeError, while older PHP versions silently converted it to 0.

SuiteCRM's legacy code still assumes PHP 5/7 behavior — so this:

return 0 + $value;


will crash if $value is not a numeric string.

✅ Why it fails

0 + $value in PHP 8 causes:

"123" → 123 ✔️

"12.5" → 12.5 ✔️

"00123" → 123 ✔️

"abc" → ❌ TypeError: Unsupported operand types: int + string

And your search input is passing non-numeric strings into quoteType() for numeric fields.

SuiteCRM never expected PHP 8 strict typing here.

🚀 Recommended final code block
```
if ($this->isNumericType($type)) {
    if (is_numeric($value)) {
        return $value + 0;   // numeric conversion
    }
    return 0;                // fallback for PHP 8 safety
}
```

### Steps to Reproduce the Issue

```bash
Wasn't able to reproduce this on another system.
One idea: Maybe it's related to non numeric values in the listview?
```

### Context

Found sth. similar in the forums, but a different code position:
https://community.suitecrm.com/t/suitecrm-7-12-x-7-13-x-php-8-0-x/87395/7

### Version

8.9.1

### What browser are you currently using?

Firefox

### Browser Version

_No response_

### Environment Information

MariaDB, PHP8.3

### Operating System and Version

Ubuntu 24

--- a/public/legacy/include/portability/classicview_routing_exclusions.php
+++ b/public/legacy/include/portability/classicview_routing_exclusions.php
@@ -41,6 +41,9 @@ $classicview_routing_exclusions = [
     ],
     'Users' => [
         'ChangePassword'
+    ],
+    'Home' => [
+        'UnifiedSearch',
     ]
 ];
 

